sisl/geom/nanoribbon.py

```diff
@@ -154,3 +159,461 @@ def zgnr(width, bond=1.42, atoms=None):
     agnr : armchair graphene nanoribbon
     """
     return graphene_nanoribbon(width, bond, atoms, kind='zigzag')
+
+
+Heteroribbon_section = namedtuple("Section", ["W", "L", "shift", "atoms"], defaults=(None, 1, 0, None))
+
+
+@set_module("sisl.geom")
+def heteroribbon(sections, bond, atoms=None, kind="armchair"):
+    """Builds a nanoribbon with multiple widths.
+
+    The function makes sure that only valid ribbons, that is
+    ribbons with no dangling bonds, are being produced.
+    Explanatory errors are raised if that's not possible. Only the 
+    connection at the periodic boundary remains unchecked.
+
+    It only works for armchair ribbons for now.
+
+    Parameters
+    ----------
+    sections: array-like of (Heteroribbon_section or array-like)
```

Comment:
instead of `Heteroribbon_section` write `heteroribbon.Section` or a `tuple` (which is equally retrievable) to backreference the attribute.